experimental/table/server/master/src/main/java/alluxio/master/catalog/Table.java

```diff
@@ -14,31 +14,62 @@
 import alluxio.grpc.catalog.ColumnStatisticsInfo;
 import alluxio.grpc.catalog.Schema;
 import alluxio.grpc.catalog.TableInfo;
+import alluxio.grpc.catalog.UdbTableInfo;
+import alluxio.proto.journal.Catalog;
 import alluxio.table.common.udb.UdbTable;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 import java.io.IOException;
-import java.util.ArrayList;
 import java.util.List;
 import java.util.stream.Collectors;
 
 /**
  * The table implementation which manages all the versions of the table.
  */
 public class Table {
-  private final String mName;
+  private static final Logger LOG = LoggerFactory.getLogger(Table.class);
+
+  private String mName;
   private final Database mDatabase;
-  private final Schema mSchema;
-  private final UdbTable mUdbTable;
+  private Schema mSchema;
+  private UdbTableInfo mTableInfo;
```

Comment:
why is this not `UdbTable`?

Reply:
my thinking was to decouple `UdbTable` from Table object so that we can't accidentally modify it. The only way to update a table object is to go through sync.